### PR TITLE
vscode api: introduce accessibilityInformation

### PR DESCRIPTION
**Pull request by isidorn**
https://github.com/microsoft/vscode/issues/95360

This PR:
* Introduces `AccessibilityInformation` to the vscode api
* Uses and adopts it for the TreeItem, StatusBarItem, TimelineItem
* Introduces `IAccessibilityInformation` interface on the vscode side to be shared

This PR is not yet tested. I might have missed some step in the endless conversion of objects. 

Alternative name to `AccessibilityInformation` is just `Accessibility`, but for some reason I prefer the longer version.

Assigning @jrieken for Api review, @alexr00 for treeItem review
fyi @eamodio @bpasero (you do not have to review if busy)

### CODE REVIEW

File: src/vs/vscode.proposed.d.ts [hunk 1]
```diff
@@ -1100,6 +1105,11 @@ declare module 'vscode' {
 			 */
 			name: string;
 
+			/**
```
Comment (jrieken) [inline src/vs/vscode.proposed.d.ts:1108]: unrelated to this PR but you could fix the typing being nested inside a namespace move it out to where it belongs

**Comment by jrieken** [comment]
fyi - for changes in vscode.proposed.d.ts no reviews are required

**Review by alexr00 — APPROVED** [review]
`TreeItem` related changes look good, thank you for making them! I have not tested them yet, but I will make fixes if needed if I find a problem when testing.

**Comment by isidorn** [comment]
Thanks for the review.
I have verified this works good with the statusBar. Thus merging in.
@alexr00 can you please just try it out and verify all is good with the tree item.

File: src/vs/vscode.proposed.d.ts (hunk 1 above)
Comment (isidorn) [reply src/vs/vscode.proposed.d.ts:1108]: The `createStatusBarItem` also belongs to the `window` namespace in `vscode.d.ts`. Thus we have to keep it in the same namespace. I could only move `StatusBarItemOptions` out to  `vscode`.

**Pull request merged**